### PR TITLE
Add `receive` fn to `HypNative` and `Scaled` variant

### LINKED ISSUE #2622: Allow ETH to be donated to a HypNative contract
The plan for bridging into Nautilus so that bridging gives a user the native Neon EVM token (i.e. ETH on that EVM) involves sending a bunch of ETH tokens to the HypNative contract without actually bridging the tokens. This ETH serves as a liquidity source for bridging *in* to that native warp route.

Atm, there is no fallback function on HypNative and the only payable function is `transferRemote` itself. We should add a function `function donate() external payable;` that anyone can call to donate funds to the HypNative contract

See https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2621 for what this looks like in the stopgap version - need to add tests etc

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Adds an extension to `HypNative` that scales `TokenRouter` message `amount` by configurable `scale`

### Drive-by changes

Adds forge to token package

### Related issues

- Fixes #2622 

### Backward compatibility

Yes

### Testing

Unit Tests

### CODE REVIEW

File: typescript/token/contracts/extensions/HypNativeScaled.sol [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {HypNative} from "../HypNative.sol";
+import {TokenRouter} from "../libs/TokenRouter.sol";
+
+/**
+ * @title Hyperlane Native Token that scales native value by a fixed factor for consistency with other tokens.
```
Comment (nambrot) [inline typescript/token/contracts/extensions/HypNativeScaled.sol:8]: Given how complicated it was for us to reason about the direction of the scale, I think a bit more documentation here would be appreciated

File: typescript/token/test/HypNativeScaled.t.sol [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        synth = new HypERC20(decimals);
+        synth.initialize(
+            address(environment.mailboxes(synthDomain)),
+            address(environment.igps(synthDomain)),
+            synthSupply,
+            "Zebec BSC Token",
+            "ZBC"
+        );
+
+        native = new HypNativeScaled(scale);
```
Comment (nambrot) [inline typescript/token/test/HypNativeScaled.t.sol:48]: This is assuming 18 decimals for the token, and i would write the test cases to be `amount`/user-focused

**Comment by avious00** [comment]
@yorhodes bump to merge this

**Review by jmrossy — COMMENTED** [review]
Only skimmed the ts changes but lgtm

**Pull request merged**